caiman/motion_correction.py

```diff
@@ -2892,10 +2898,10 @@ def motion_correct_batch_rigid(fname, max_shifts, dview=None, splits=56, num_spl
 
         fname_tot_rig, res_rig = motion_correction_piecewise(fname, splits, strides=None, overlaps=None,
                                                              add_to_movie=add_to_movie, template=old_templ, max_shifts=max_shifts, max_deviation_rigid=0,
-                                                             dview=dview, save_movie=save_movie, base_name=base_name, subidx = subidx,
+                                                             dview=dview, save_movie=save_movie, base_name=base_name,
                                                              num_splits=num_splits_to_process, shifts_opencv=shifts_opencv, nonneg_movie=nonneg_movie, gSig_filt=gSig_filt,
                                                              use_cuda=use_cuda, border_nan=border_nan, var_name_hdf5=var_name_hdf5, is3D=is3D,
-                                                             indices=indices)
+                                                             indices=indices, subidx=None)
```

Comment:
FYI subidx=None by default in motion_correction_piecewise, but no harm in stating that explicitly here

Reply:
Right, and I don't know why they had `subidx=subidx` in the function call, as it tells motion correction to run just on the specified slice (that's what was making `subidx` dual function, one the one hand to constrict frames from which to compute the template, and on the other hand by computing motion correction on this only). Writing this I think they maybe never had the use case of wanting to apply a template on a superset of frames, if it makes sense! So maybe they just put that here to say "if I have a movie and just want to do MC from frame A to frame B, I can do so" and it impacted both template computation and MC itself.